cmd/vicadmin/usersession.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"sync"
+	"time"
+)
+
+// UserSession holds a user's session metadata
+type UserSession struct {
+	username string
+	created  time.Time
+	config   *session.Config
+}
+
+// UserSessionStore holds and manages user sessions
+type UserSessionStore struct {
+	mutex    sync.RWMutex
+	sessions map[string]*UserSession
+	ticker   *time.Ticker
+}
+
+type UserSessionStorer interface {
+	Add(username string, config *session.Config) *UserSession
+	Delete(username string)
+	GetRealSession(username string) (vSphereSession *session.Session, err error)
+	GetUserSession(username string) *UserSession
+}
+
+func (u *UserSessionStore) Add(username string, config *session.Config) *UserSession {
+	u.mutex.Lock()
+	defer u.mutex.Unlock()
+	u.sessions[username] = &UserSession{
```

Comment:
@vburenin pointed this out before; you can store the new `UserSession` in a local variable to avoid the second lookup on line 52.